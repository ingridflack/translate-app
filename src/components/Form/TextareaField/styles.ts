import styled from "styled-components";

export const Container = styled.div`
  width: 100%;
`;

export const Textarea = styled.textarea`
  width: 100%;
  padding: 0;
  resize: none;
  border: none;
  outline: none;
  font-size: 24px;
`;
