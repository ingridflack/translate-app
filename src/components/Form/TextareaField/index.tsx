import { TextareaHTMLAttributes } from "react";
import { Container, Textarea } from "./styles";

interface TextareaFieldProps
  extends TextareaHTMLAttributes<HTMLTextAreaElement> {
  rows?: number;
  cols?: number;
  text?: string;
}
const TextareaField = ({ rows = 5, cols, ...props }: TextareaFieldProps) => {
  return (
    <Container>
      <Textarea rows={rows} cols={cols} {...props} />
    </Container>
  );
};

export default TextareaField;
