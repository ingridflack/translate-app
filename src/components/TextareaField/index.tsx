import { Container, Textarea } from "./styles";

interface TextareaFieldProps {
  rows?: number;
  cols?: number;
}
const TextareaField = ({ rows = 10, cols = 60 }: TextareaFieldProps) => {
  return (
    <Container>
      <Textarea rows={rows} cols={cols} />
    </Container>
  );
};

export default TextareaField;
