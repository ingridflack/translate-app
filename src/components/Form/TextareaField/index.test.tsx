import { render } from "@testing-library/react";

import TextareaField from ".";

describe("<TextareaField/>", () => {
  it("should render correctly", () => {
    const { container } = render(<TextareaField />);

    expect(container).toMatchSnapshot();
  });
});
