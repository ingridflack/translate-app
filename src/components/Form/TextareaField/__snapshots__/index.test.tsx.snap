// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextareaField/> should render correctly 1`] = `
<div>
  <div
    class="sc-bdvvtL eukWAO"
  >
    <textarea
      class="sc-gsDKAQ coeeJH"
      rows="5"
    />
  </div>
</div>
`;
